File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int digits(long num);

int main(void)
{
    long card_num = get_long("Enter your credit card number :");
    int digit = digits(card_num);
    if (digit != 13 && digit != 15 && digit != 16)
    {
        printf("INVALID\n");
        return -1;
    }
    else
    {
        long card_cpy = card_num;
        printf("%li\n",card_cpy);
        int remainder = 0;
        int  k = 0, sum1 = 0, sum2 = 0;;
        while (card_cpy > 0)
        {
            remainder = card_cpy % 10;
            card_cpy = card_cpy / 10;
            if (k)
            {
                while (remainder > 0)
                {
                    
                    remainder = remainder / 10;

                }
            }
            else
            {
                sum2 += remainder;
                k = 1;
            }
        }
        printf("%d\n",sum1);
        printf("%d\n",sum2);

    }

}

// calculate the number of digits in a given number.
int digits(long num)
{
    int dig = 0;
    while (num > 0)
    {
        num = num /10;
        dig++;
    }
    return dig;
}

File: sentimental-credit/credit.py
# TODO
from cs50 import get_int, get_string
import re

def main():
    number = get_int("Number: ")
    num_list = re.findall('[0-9]', str(number))
    digits = len(num_list)

    if digits != 13 and digits != 15 and digits != 16:
        print(INVALID)
        return 0
    else:
        first = int(num_list[0])
        sec = int(num_list[1])
        card_cpy = card_num
        remainder, rem = 0, 0
        k, sum1, sum2, total = 0, 0, 0, 0

        while card_cpy > 0:
            remainder = card_cpy % 10
            card_cpy /= 10
            if k:
                remainder *= 2
                while remainder > 0:
                    rem = remainder % 10
                    remainder /= 10
                    sum1 += rem
                k = 0
            else:
                sum2 += remainder
                k = 1

            else
            {
                sum2 += remainder;
                k = 1;
            }
        }
        total = sum1 + sum2;
        // validate the card :
        if ((total % 10) != 0)
        {
            printf("INVALID\n");
            return 0;
        }
        else if (first == 4)
        {
            printf("VISA\n");
            return 0;
        }
        else if (two == 34 || two == 37)
        {
            printf("AMEX\n");
            return 0;
        }
        else if (two == 51 || two == 52 || two == 53 || two == 54 || two == 55)
        {
            printf("MASTERCARD\n");
            return 0;
        }
        else
        {
            printf("INVALID\n");
            return 0;
        }

